src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,10 +31,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+        String[] fullAddress = address.split(", ");
+        this.block = new Block(fullAddress[0].trim());
```

Comment:
- What if all 4 component are not entered by the user?

- Did you run the regression? It is a must to run the regression and change testcases if needed with these changes.